Create _7_21_1_Hermite_Polynomials

wg030 opened the pull request:
I am currently working on a project where I need to evaluate hermite polynomials. I cam accross this GSL package, but I have figured out that for whatever reason there is no wrapper for them.
Altough I am quite new to julia i tried my best to write the wrapper on my own by looking at the code of functions that have already been implemented.

yuyichao commented on src/_7_21_1_Hermite_Polynomials.jl:
```diff
@@ -0,0 +1,140 @@
+function sf_hermite_prob(n::Integer, x::Real)
+    ccall( (:gsl_sf_hermite_prob, libgsl), Cdouble, (Cint, Cdouble), n, x )
+end
+Compat.@dep_vectorize_2arg Number sf_hermite_prob
```
New functions shouldn't have this.

wg030 replied:
yuyichao, what exactly do you mean with "New functions shouldn't have this."
Are you referring to the Compat line behind the functions?!
As far as I understand their job is to ensure compatibility with older versions of julia. As I said I simply copied them from other wrappers in the library. However if they are not necessary anymore I will remove them, of course.

yuyichao replied:
> Are you referring to the Compat line behind the functions?!

Yes.

> As far as I understand their job is to ensure compatibility with older versions of julia.

No. It's for compatibility with users that haven't upgraded as explained in the doc of the Compat package https://github.com/JuliaLang/Compat.jl#new-macros

> As I said I simply copied them from other wrappers in the library.

I know that's why I mentioned "New functions".

wg030 replied:
Ok, thanks a lot for the clearification!
I will update the file accordingly.

wg030 commented:
I have now nearly figured out how to test my file.
I simpliy put it into the folder "\.julia\v0.6\GSL\src", added a further line to "__FILELIST.jl", then called "Pkg.build()" in julia. When I wanted to run a tesfile, julia started to precompile "GSL" again. I got some errors, which where syntax erros that I have corrected now. However after successfully precompiling it couldn't still call the functions. It throw the error "LoadError: ccall: could not find function gsl_sf_hermite_phys in library \.julia\v0.6\WinRPM\deps\usr\x86_64-w64-mingw32\sys-root\mingw\bin\libgsl-0.DLL", which seems pretty strange to me.
Having said this, I have successfully precomplied the code, but I haven't testest the results yet.

wg030 commented:
I have now figured out that the "libgsl-0.dll" of WinRPM only contains the functions for which a wrapper has already been added to GSL package. Since my wrapper for the Hermite Polynomials has not been added yet the error above occured.

In order to test my wrapper I compiled the whole GNU Scientific Library on my notebook using the programms MinGW-w64 and MYSYS. So I replaced the "libgsl-0.dll" and also "libgslcblas-0.dll" from the  .julia\v0.6\WinRPM\deps\usr\x86_64-w64-mingw32\sys-root\mingw\bin directory with the dlls that I compiled myself.
This worked out so that I could test my file now. While testing I turned out that the implementation of the two functions "sf_hermite_prob_array" and "sf_hermite_phys_array" were not correct. I have found the misstake though and successfully tested all functions. I verified the reults with the help of MAPLE's HermiteH function and it seemed that everything works as it should.

I will now upload the final version of my wrapper for the Hermite Polynomilas as well as the testfile that I wrote and I hope that my pull request will be merged soon.

ludvigak commented:
Hermite polynomials are now available in the master branch, since we upgraded to GSL 2.5. I'm sorry your PR never got merged.
